packages/node-resolve/src/util.js

```diff
@@ -144,7 +144,18 @@ export function getPackageInfo(options) {
     if (typeof packageSideEffects === 'boolean') {
       internalPackageInfo.hasModuleSideEffects = () => packageSideEffects;
     } else if (Array.isArray(packageSideEffects)) {
-      internalPackageInfo.hasModuleSideEffects = createFilter(packageSideEffects, null, {
+      let finalPackageSideEffects = packageSideEffects;
+      finalPackageSideEffects = packageSideEffects.map((sideEffect) => {
```

Comment:
Please could you make this `const final…` and initialise the variable here, then it could be one line not 2